### PR TITLE
ci: Publish ESLint results for builds on ci.jenkins.io

### PR DESCRIPTION
**Pull request by NotMyFault**
Warnings ng, more specifically, analysis-model, is able to render ESLint warnings and results on the web UI:

<details>
<summary>Issues recorded</summary>

Screenshot 2022-07-08 at 23 11 50

</details>

<details>
<summary>Job overview</summary>

Screenshot 2022-07-08 at 22 24 32

</details>

The overview is pretty handy, if you want to view failures and warnings outside the build log in a more appealing way.

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6817"></a>

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

File: Jenkinsfile
Comment (timja) [inline Jenkinsfile:120]: new is almost never what you want, it's not reproduce-able outside the CI system
```suggestion
                [threshold: 1, type: 'TOTAL', unstable: true],
```

**Review by timja — APPROVED** [review]

File: war/package.json
Comment (basil) [inline war/package.json:18]: I think using the shell redirection operator here is a bit fragile, as it assumes this command will be executed by a shell, which seems to be the case in the existing implementation but is not guaranteed to be the case. Better to use an explicit command-line argument which will work whether this command is executed by a shell or not:

```suggestion
    "lint:js": "eslint . --ext js -f checkstyle -o target/eslint-warnings.xml",
```

**Review by basil — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

File: war/package.json
Comment (basil) [inline war/package.json:18]: Caused [JENKINS-71021](https://issues.jenkins.io/browse/JENKINS-71021).

**Comment by mawinter69** [comment]
There is another problem with this change. When we have an eslint error the maven build will fail stopping the pipeline so the recordIssues step is never reached.

**Pull request merged**
